File: Daily Test/Java/Count Set Bits - All Positions 20.3.2026.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];

        for(int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        String[] b = new String[n];
        int le = 0;

        for(int i = 0; i < n; i++) {
            String s = Integer.toBinaryString(arr[i]);
            b[i] = s;
            le = Math.max(le, s.length());
        }

        int lsb = -1;

        while(le >= -lsb) {
            int ans = 0;

            for(int i = 0; i < n; i++) {
                if(b[i].length() >= -lsb) {
                    if(b[i].charAt(b[i].length() + lsb) == '1') {
                        ans++;
                    }
                }
            }

            lsb--;
            System.out.print(ans + " ");
        }
    }
}

File: Daily Test/C++/Count Set Bits - All Positions 20.3.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> arr(n);
    for(int i = 0; i < n; i++) cin >> arr[i];

    vector<string> b;
    int le = 0;

    for(int x : arr) {
        string s = "";
        while(x > 0) {
            s = char((x % 2) + '0') + s;
            x /= 2;
        }
        if(s == "") s = "0";

        le = max(le, (int)s.size());
        b.push_back(s);
    }

    int lsb = -1;

    while(le >= -lsb) {
        int ans = 0;

        for(int i = 0; i < n; i++) {
            if((int)b[i].size() >= -lsb) {
                if(b[i][b[i].size() + lsb] == '1') {
                    ans++;
                }
            }
        }

        lsb--;
        cout << ans << " ";
    }

    return 0;
}